File: Algorithms/111.minimum-depth-of-binary-tree/minimum-depth-of-binary-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
class Solution {
    public int minDepth(TreeNode root) {
        if (root == null) return 0;

        int leftDepth = minDepth(root.left);
        int rightDepth = minDepth(root.right);

        // current node is not leaf node
        if (root.left == null) {
            return 1 + rightDepth;
        } else if (root.right == null) {
            return 1 + leftDepth;
        }

        return 1 + Math.min(leftDepth, rightDepth);
    }
}

File: Algorithms/111.minimum-depth-of-binary-tree/minimum-depth-of-binary-tree.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
 func minDepth(root *TreeNode) int {
    if root == nil {
        return 0
    }
    leftDepth := minDepth(root.Left)
    rightDepth := minDepth(root.Right)
    if root.Left == nil {
        return 1 + rightDepth
    }else if root.Right == nil {
        return 1 + leftDepth
    }

    return 1 + min(leftDepth,rightDepth)
}

func min(a,b int)int {
    if a < b {
        return a
    } else {
        return b
    }
}